src/lib.rs

```diff
@@ -365,6 +368,8 @@ pub enum Notification {
         // I assume that any decent notification daemon will handle an invalid ID
         // value correctly, but this code should probably test for this at the start
         // so that it cannot be used with a server that crashes in this case.
+        // marmarek: validate if the notification ID comes from this client (do not allow replacing
+        // notifications sent by other qubes)
```

Comment:
Should this use the portal (`org.freedesktop.portal.Notification` interface) instead?

Reply:
Looking at the API, I don't think it's a good idea. While it seems to handle the ID mapping issue, there are enough differences that other issues probably will appear. For example, there is no info when notification is closed (and application may expect to receive this signal), or body formatting differences (the portal spec is very unclear about that).